### ISSUE TITLE
New Rule: no-top-level-await

### ISSUE THREAD

**Issue by DMartens** [eslint-community/eslint-plugin-n#439]
### Rule details

It would warn about using top-level awaits as it conflicts with `require(esm)`.
Unfortunately `no-restricted-syntax` cannot be used as the following should be ignored:
- files with a shebang or without exports
- files called from `package.json` scripts 

### What type of rule is this?

Warns about a potential problem

### Example code

```js
await fn();

for await (const x of asyncGenerator()) {
}
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

This rule was discussed in the eslint repo but [declined with a reference to open here]((https://github.com/eslint/eslint/issues/19481#issuecomment-2714730967).

**Comment by scagood** [comment]
I like this idea, I think it'd be best to keep it reasonably simple at the start though (eg detecting cjs/esm and erroring only for cjs)

`ignores` is always a nice option to have :)

**Comment by voxpelli** [comment]
It's in ESM files it should error, as any ESM file with a top level await can't be used with the new ESM-support in `require()` 🙂

Maybe there should be an (optional) exception for CLI-scripts, as they will never be used with `require()` and top level await is quite helpful in those
